src/general_tally.cpp

```diff
@@ -0,0 +1,210 @@
+#include <tallies/general_tally.hpp>
+#include <utils/error.hpp>
+
+GeneralTally::GeneralTally(std::shared_ptr<PositionFilter> position_filter,
+                           std::shared_ptr<EnergyFilter> energy_in,
+                           Quantity quantity, Estimator estimator,
+                           std::string name_)
+    : ITally(quantity, estimator, name_),
+      position_filter_(position_filter),
+      energy_in_(energy_in) {
+  // flag variable to predict weather position and energy filter exists or not
+  bool check_ifany_filter = true;
+
+  // get the shape or dimension of position filter
+  StaticVector3 position_shape_;
+  if (position_filter_) {
+    position_shape_ = position_filter_->get_shape();
+    check_ifany_filter = false;
+  }
+
+  StaticVector4 tally_dimensions_(position_shape_.begin(),
+                                  position_shape_.end());
+
+  // get the size of the enery_filter
+  if (energy_in_) {
+    size_t ne = energy_in_->size();
+    tally_dimensions_.insert(tally_dimensions_.begin(), ne);
+
+    check_ifany_filter = check_ifany_filter && false;
+  }
+
+  // if any of the filter is not present, then give fatal_error
+  if (check_ifany_filter == true) {
+    fatal_error("for the " + name_ + " tally, no filter is provided.");
+  }
```

Comment:
Isn't it easier to simply call fatal error at the beginning if `energy_in_ == nullptr  && position_filter_ == nullptr` ?